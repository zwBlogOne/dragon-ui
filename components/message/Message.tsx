import React, { Component } from 'react';
import MessageItem from './MessageItem';
import PropsType from './PropsType';

class Message extends Component<PropsType, any> {
  static defaultProps = {
    prefixCls: 'ui-message',
    msg: [],
    duration: 1500,
    theme: 'info',
  };

  static Item;

  private timer;
  private active = 0;
  private message;

  componentWillReceiveProps() {
    clearTimeout(this.timer);
    this.enter();
  }

  enter() {
    const { prefixCls } = this.props;
    this.timer = setTimeout(() => {
      const node = this.message.children;

      if (!node[this.active]) {
        return;
      }

      node[this.active].className = `${prefixCls}-wrapper`;

      this.active++;

      if (this.active !== this.message.children.length) {
        this.enter();
      }
    }, this.props.duration);
  }

  render() {
    const {
      msg, duration, theme, style, prefixCls,
    } = this.props;

    // eslint-disable-next-line
    let items = msg.map((o, i) => {
      return (
        // eslint-disable-next-line
        <MessageItem key={i} content={o.m} duration={duration} theme={theme} />
      );
    });

    return (
      <div
        className={prefixCls}
        ref={(message) => { this.message = message; }}
        style={style}
      >
        {items}
      </div>
    );
  }
}

export default Message;
