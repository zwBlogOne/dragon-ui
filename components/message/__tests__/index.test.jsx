import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Message from '../index';

describe('Message', () => {
  it('renders message correctly', () => {
    const wrapper = mount(
      <Message
        msg={[
          {
            m: '提交成功',
          },
        ]}
        theme="info"
        duration={5000}
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
