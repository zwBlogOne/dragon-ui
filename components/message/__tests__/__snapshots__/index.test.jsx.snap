// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message renders message correctly 1`] = `
<MessageExtension
  duration={5000}
  msg={
    Array [
      Object {
        "m": "提交成功",
      },
    ]
  }
  prefixCls="ui-message"
  theme="info"
>
  <div
    className="ui-message"
  >
    <MessageItem
      content="提交成功"
      duration={5000}
      key="0"
      prefixCls="ui-message"
      theme="info"
    >
      <div
        className="ui-message-wrapper"
      >
        <div
          className="ui-message-container"
        >
          <Icon
            className="ui-message-info"
            onClick={[Function]}
            prefixCls="ui-icon"
            style={
              Object {
                "fontSize": 15,
              }
            }
            theme="default"
            type="info-round"
          >
            <i
              className="ui-icon ui-icon-info-round theme-default ui-message-info"
              onClick={[Function]}
              style={
                Object {
                  "fontSize": 15,
                }
              }
            />
          </Icon>
          提交成功
        </div>
      </div>
    </MessageItem>
  </div>
</MessageExtension>
`;
